File: BitManipulation/LeetCode/Easy/PowerOf2/PowerOf2.java
// https://leetcode.com/problems/power-of-two/description/
public class PowerOf2 {
    public static boolean isPowerOfTwo(int n) {
        return ((n > 0) && (n & (n - 1)) == 0);
    }
    public static void main(String[] args) {
        System.out.println(isPowerOfTwo(64));
    }
}

File: BitManipulation/LeetCode/Easy/PowerOf2/PowerOf2.cpp
// https://leetcode.com/problems/power-of-two/description/
#include<bits/stdc++.h>
using namespace std;
bool isPowerOfTwo(int n)
{
    // remove the last set bit and check is the number is 0 or not
    return (n > 0 && !(n & n - 1));
}
int main(void) {
    cout << isPowerOfTwo(45) << "\n" << isPowerOfTwo(32);
    return 0;
}